### PR TITLE
Make logviewer aware of mergability failures

### PR DESCRIPTION
**Pull request by PaliC**
Make logviewer aware of mergability failures

Adds a rule/some text to help the logviewer understand mergability errors

### CODE REVIEW

File: aws/lambda/log-classifier/ruleset.toml [hunk 1]
```diff
@@ -256,6 +256,10 @@ pattern = 'IMPROVED:\s+graph_breaks='
 name = 'npm error'
 pattern = '^npm ERR! code .*'
 
+[[rule]]
+name = 'ghstack merge check error'
+pattern = '[Mergeability Error] .*'
```
Comment (janeyx99) [inline aws/lambda/log-classifier/ruleset.toml:261]:
```suggestion
pattern = 'is NOT mergeable into main'
```

Wouldn't this already work? It looks like the ones that require the start to match exactly use a `^`.

Comment (izaitsevfb) [reply aws/lambda/log-classifier/ruleset.toml:261]: I'd also note that `[]` are special characters in regex and need to be escaped.

Comment (PaliC) [reply aws/lambda/log-classifier/ruleset.toml:261]: Will just remove the `[]`. I rather just name the error and match against that as generally the content of an error should fungible imo

**Review by izaitsevfb — CHANGES_REQUESTED** [review]
Rule check failed:
https://github.com/pytorch/test-infra/actions/runs/7548211162/job/20549812678?pr=4888

**Review by izaitsevfb — APPROVED** [review]

**Pull request merged**